Subject: [PATCH] Match SDK37 dialogs and fragment-owned main switches

Keep the original UN1CA screens, preferences and branding. Use the actual
SDK37 AppCompat signatures and native switch/content layout.

diff --git a/smali_classes4/io/mesalabs/unica/utils/Utils.smali b/smali_classes4/io/mesalabs/unica/utils/Utils.smali
--- a/smali_classes4/io/mesalabs/unica/utils/Utils.smali
+++ b/smali_classes4/io/mesalabs/unica/utils/Utils.smali
@@ -37,7 +37,7 @@
 
     move-result v1
 
-    invoke-virtual {v0, v1}, Landroidx/appcompat/app/AlertDialog$Builder;->setTitle(I)Landroidx/appcompat/app/AlertDialog$Builder;
+    invoke-virtual {v0, v1}, Landroidx/appcompat/app/AlertDialog$Builder;->setTitle(I)V
 
     const-string v1, "unica_dialog_reboot_msg"
 
@@ -45,7 +45,7 @@
 
     move-result v1
 
-    invoke-virtual {v0, v1}, Landroidx/appcompat/app/AlertDialog$Builder;->setMessage(I)Landroidx/appcompat/app/AlertDialog$Builder;
+    invoke-virtual {v0, v1}, Landroidx/appcompat/app/AlertDialog$Builder;->setMessage(I)V
 
     const-string v1, "unica_dialog_reboot_positive"
 
@@ -57,7 +57,7 @@
 
     invoke-direct {v3, p0}, Lio/mesalabs/unica/utils/Utils$$ExternalSyntheticLambda0;-><init>(Landroid/content/Context;)V
 
-    invoke-virtual {v0, v1, v3}, Landroidx/appcompat/app/AlertDialog$Builder;->setPositiveButton(ILandroid/content/DialogInterface$OnClickListener;)Landroidx/appcompat/app/AlertDialog$Builder;
+    invoke-virtual {v0, v1, v3}, Landroidx/appcompat/app/AlertDialog$Builder;->setPositiveButton(ILandroid/content/DialogInterface$OnClickListener;)V
 
     const-string p0, "unica_dialog_reboot_negative"
 
@@ -67,7 +67,7 @@
 
     const/4 v1, 0x0
 
-    invoke-virtual {v0, p0, v1}, Landroidx/appcompat/app/AlertDialog$Builder;->setNegativeButton(ILandroid/content/DialogInterface$OnClickListener;)Landroidx/appcompat/app/AlertDialog$Builder;
+    invoke-virtual {v0, p0, v1}, Landroidx/appcompat/app/AlertDialog$Builder;->setNegativeButton(ILandroid/content/DialogInterface$OnClickListener;)V
 
     invoke-virtual {v0}, Landroidx/appcompat/app/AlertDialog$Builder;->create()Landroidx/appcompat/app/AlertDialog;
 
@@ -151,6 +151,17 @@
 
     sget-object v0, Lcom/android/settings/overlay/FeatureFactoryImpl;->_appContext:Lcom/android/settings/SettingsApplication;
 
+    if-nez v0, :cond_appctx_ok
+
+    new-instance v0, Ljava/lang/NullPointerException;
+
+    const-string v1, "FeatureFactoryImpl._appContext is null"
+
+    invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
+
+    throw v0
+
+    :cond_appctx_ok
     invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
 
     move-result-object v0

diff --git a/smali_classes4/io/mesalabs/unica/settings/hma/HideMyApplistFragment.smali b/smali_classes4/io/mesalabs/unica/settings/hma/HideMyApplistFragment.smali
--- a/smali_classes4/io/mesalabs/unica/settings/hma/HideMyApplistFragment.smali
+++ b/smali_classes4/io/mesalabs/unica/settings/hma/HideMyApplistFragment.smali
@@ -249,14 +249,56 @@
 
     iput-object v1, p0, Lio/mesalabs/unica/settings/hma/HideMyApplistFragment;->mLoadingViewController:Lcom/android/settings/widget/LoadingViewController;
 
-    invoke-virtual {p0}, Landroidx/fragment/app/Fragment;->getActivity()Landroidx/fragment/app/FragmentActivity;
+    const-string p1, "layout"
 
+    const-string p2, "sec_settings_main_prefs"
+
+    invoke-static {p1, p2}, Lio/mesalabs/unica/utils/Utils;->getResourceId(Ljava/lang/String;Ljava/lang/String;)I
+
+    move-result p1
+
+    iget-object p2, p0, Lio/mesalabs/unica/settings/hma/HideMyApplistFragment;->mContext:Landroid/content/Context;
+
+    invoke-static {p2}, Landroid/view/LayoutInflater;->from(Landroid/content/Context;)Landroid/view/LayoutInflater;
+
+    move-result-object p2
+
+    const/4 v0, 0x0
+
+    invoke-virtual {p2, p1, v0}, Landroid/view/LayoutInflater;->inflate(ILandroid/view/ViewGroup;)Landroid/view/View;
+
     move-result-object p1
 
-    check-cast p1, Lcom/android/settings/SettingsActivity;
+    const-string p2, "id"
 
-    iget-object p1, p1, Lcom/android/settings/SettingsActivity;->mMainSwitch:Lcom/android/settings/widget/SettingsMainSwitchBar;
+    const-string v0, "main_content"
 
+    invoke-static {p2, v0}, Lio/mesalabs/unica/utils/Utils;->getResourceId(Ljava/lang/String;Ljava/lang/String;)I
+
+    move-result v0
+
+    invoke-virtual {p1, v0}, Landroid/view/View;->requireViewById(I)Landroid/view/View;
+
+    move-result-object v0
+
+    check-cast v0, Landroid/view/ViewGroup;
+
+    invoke-virtual {v0, p3}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
+
+    move-object p3, p1
+
+    const-string v0, "switch_bar"
+
+    invoke-static {p2, v0}, Lio/mesalabs/unica/utils/Utils;->getResourceId(Ljava/lang/String;Ljava/lang/String;)I
+
+    move-result p2
+
+    invoke-virtual {p3, p2}, Landroid/view/View;->requireViewById(I)Landroid/view/View;
+
+    move-result-object p1
+
+    check-cast p1, Lcom/android/settings/widget/SettingsMainSwitchBar;
+
     iput-object p1, p0, Lio/mesalabs/unica/settings/hma/HideMyApplistFragment;->mSwitchBar:Lcom/android/settings/widget/SettingsMainSwitchBar;
 
     iget-object p2, p0, Lio/mesalabs/unica/settings/hma/HideMyApplistFragment;->mContext:Landroid/content/Context;

diff --git a/smali_classes4/io/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment.smali b/smali_classes4/io/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment.smali
--- a/smali_classes4/io/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment.smali
+++ b/smali_classes4/io/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment.smali
@@ -249,14 +249,56 @@
 
     iput-object v1, p0, Lio/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment;->mLoadingViewController:Lcom/android/settings/widget/LoadingViewController;
 
-    invoke-virtual {p0}, Landroidx/fragment/app/Fragment;->getActivity()Landroidx/fragment/app/FragmentActivity;
+    const-string p1, "layout"
 
+    const-string p2, "sec_settings_main_prefs"
+
+    invoke-static {p1, p2}, Lio/mesalabs/unica/utils/Utils;->getResourceId(Ljava/lang/String;Ljava/lang/String;)I
+
+    move-result p1
+
+    iget-object p2, p0, Lio/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment;->mContext:Landroid/content/Context;
+
+    invoke-static {p2}, Landroid/view/LayoutInflater;->from(Landroid/content/Context;)Landroid/view/LayoutInflater;
+
+    move-result-object p2
+
+    const/4 v0, 0x0
+
+    invoke-virtual {p2, p1, v0}, Landroid/view/LayoutInflater;->inflate(ILandroid/view/ViewGroup;)Landroid/view/View;
+
     move-result-object p1
 
-    check-cast p1, Lcom/android/settings/SettingsActivity;
+    const-string p2, "id"
 
-    iget-object p1, p1, Lcom/android/settings/SettingsActivity;->mMainSwitch:Lcom/android/settings/widget/SettingsMainSwitchBar;
+    const-string v0, "main_content"
 
+    invoke-static {p2, v0}, Lio/mesalabs/unica/utils/Utils;->getResourceId(Ljava/lang/String;Ljava/lang/String;)I
+
+    move-result v0
+
+    invoke-virtual {p1, v0}, Landroid/view/View;->requireViewById(I)Landroid/view/View;
+
+    move-result-object v0
+
+    check-cast v0, Landroid/view/ViewGroup;
+
+    invoke-virtual {v0, p3}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
+
+    move-object p3, p1
+
+    const-string v0, "switch_bar"
+
+    invoke-static {p2, v0}, Lio/mesalabs/unica/utils/Utils;->getResourceId(Ljava/lang/String;Ljava/lang/String;)I
+
+    move-result p2
+
+    invoke-virtual {p3, p2}, Landroid/view/View;->requireViewById(I)Landroid/view/View;
+
+    move-result-object p1
+
+    check-cast p1, Lcom/android/settings/widget/SettingsMainSwitchBar;
+
     iput-object p1, p0, Lio/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment;->mSwitchBar:Lcom/android/settings/widget/SettingsMainSwitchBar;
 
     iget-object p2, p0, Lio/mesalabs/unica/settings/spoof/HideDeveloperStatusFragment;->mContext:Landroid/content/Context;

diff --git a/smali_classes4/io/mesalabs/unica/search/UnicaSearchIndexableResources.smali b/smali_classes4/io/mesalabs/unica/search/UnicaSearchIndexableResources.smali
--- a/smali_classes4/io/mesalabs/unica/search/UnicaSearchIndexableResources.smali
+++ b/smali_classes4/io/mesalabs/unica/search/UnicaSearchIndexableResources.smali
@@ -1,14 +1,12 @@
 .class public final Lio/mesalabs/unica/search/UnicaSearchIndexableResources;
-.super Lcom/android/settingslib/search/SearchIndexableResourcesMobile;
+.super Ljava/lang/Object;
 .source "UnicaSearchIndexableResources.java"
 
 
 # direct methods
-.method public constructor <init>()V
+.method public static register(Lcom/android/settingslib/search/SearchIndexableResourcesBase;)V
     .locals 3
 
-    invoke-direct {p0}, Lcom/android/settingslib/search/SearchIndexableResourcesMobile;-><init>()V
-
     new-instance v0, Lcom/android/settingslib/search/SearchIndexableData;
 
     const-class v1, Lio/mesalabs/unica/settings/UnicaSettingsFragment;
@@ -17,7 +15,7 @@
 
     invoke-direct {v0, v1, v2}, Lcom/android/settingslib/search/SearchIndexableData;-><init>(Ljava/lang/Class;Lcom/android/settingslib/search/Indexable$SearchIndexProvider;)V
 
-    invoke-virtual {p0, v0}, Lio/mesalabs/unica/search/UnicaSearchIndexableResources;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
+    invoke-virtual {p0, v0}, Lcom/android/settingslib/search/SearchIndexableResourcesBase;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
 
     new-instance v0, Lcom/android/settingslib/search/SearchIndexableData;
 
@@ -27,7 +25,7 @@
 
     invoke-direct {v0, v1, v2}, Lcom/android/settingslib/search/SearchIndexableData;-><init>(Ljava/lang/Class;Lcom/android/settingslib/search/Indexable$SearchIndexProvider;)V
 
-    invoke-virtual {p0, v0}, Lio/mesalabs/unica/search/UnicaSearchIndexableResources;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
+    invoke-virtual {p0, v0}, Lcom/android/settingslib/search/SearchIndexableResourcesBase;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
 
     new-instance v0, Lcom/android/settingslib/search/SearchIndexableData;
 
@@ -37,7 +35,7 @@
 
     invoke-direct {v0, v1, v2}, Lcom/android/settingslib/search/SearchIndexableData;-><init>(Ljava/lang/Class;Lcom/android/settingslib/search/Indexable$SearchIndexProvider;)V
 
-    invoke-virtual {p0, v0}, Lio/mesalabs/unica/search/UnicaSearchIndexableResources;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
+    invoke-virtual {p0, v0}, Lcom/android/settingslib/search/SearchIndexableResourcesBase;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
 
     new-instance v0, Lcom/android/settingslib/search/SearchIndexableData;
 
@@ -47,7 +45,7 @@
 
     invoke-direct {v0, v1, v2}, Lcom/android/settingslib/search/SearchIndexableData;-><init>(Ljava/lang/Class;Lcom/android/settingslib/search/Indexable$SearchIndexProvider;)V
 
-    invoke-virtual {p0, v0}, Lio/mesalabs/unica/search/UnicaSearchIndexableResources;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
+    invoke-virtual {p0, v0}, Lcom/android/settingslib/search/SearchIndexableResourcesBase;->addIndex(Lcom/android/settingslib/search/SearchIndexableData;)V
 
     return-void
 .end method

